spec/models/application_choice_spec.rb

```diff
@@ -352,18 +352,20 @@
           .to change(application_choice, :provider_ids).to(expected_ids)
       end
 
-      it 'updates provider_ids if the course is updated' do
+      it 'updates course_option and current_course_option provider_ids if new courses are provided' do
+        new_course_option = create(:course_option, course: course)
```

Comment:
can we create a new course here to ensure we point to a different provider for `current_course_option`?